src/containers/List/ListFilters/common.tsx

```diff
@@ -10,39 +11,52 @@ import {
   ReviewerAction,
 } from '../../../utils/generated/graphql'
 import useLocalisedEnums from '../../../utils/hooks/useLocalisedEnums'
+import { FilterTitleProps } from './DateFilter/types'
 
 export const startCase = (string: string) => lodashStartCase(string.toLowerCase())
 
 const FilterContainer: React.FC<FilterContainerProps> = ({
   children,
   title = '',
-  selectedCount = 0,
+  label,
   onRemove,
-  replacementTrigger,
+  trigger,
+  setFocus = () => {},
 }) => {
   const { strings } = useLanguageProvider()
+  const [isOpen, setIsOpen] = useState(false)
+
+  const ref = useOnclickOutside(() => setIsOpen(false))
+
   return (
     <div className="active-filter">
-      {selectedCount > 0 && (
+      {label && (
         <Label color="grey" circular size="mini" floating>
-          {selectedCount}
+          {label}
         </Label>
       )}
-      <Dropdown
-        multiple
-        text={!replacementTrigger ? title : undefined}
-        trigger={replacementTrigger}
-        icon={replacementTrigger ? null : undefined}
-      >
-        <Dropdown.Menu>
-          {children}
-          <Dropdown.Divider />
-          <Dropdown.Item className="remove-filter" key="removeFilter" onClick={() => onRemove()}>
-            <Icon name="remove circle" />
-            {strings.FILTER_REMOVE}
-          </Dropdown.Item>
-        </Dropdown.Menu>
-      </Dropdown>
+      <div ref={ref}>
+        <Dropdown
+          multiple
+          text={!trigger ? title : undefined}
+          trigger={trigger && <div onClick={() => setIsOpen(!isOpen)}>{trigger}</div>}
+          icon={trigger ? null : undefined}
+          open={isOpen}
+          onOpen={() => {
+            setIsOpen(true)
+            setFocus()
+          }}
```

Comment:
This (along with the `useOnClickOutside` hook) is what allows the auto-focus text input on open to work.

If you try and auto-focus the text input, it just closes the dropdown immediately, as by focusing the text input you're "onBlur"-ing the parent.

So I switched it to a controlled component with `isOpen/setIsOpen` management. But the problem with that is that we can't use `onBlur` to detect when to close the Dropdown, since (as mentioned above), the input element steals focus and forces this to close.

Solution: use the "useOnClickOutside" hook to detect when clicking outside the whole component, then `setIsOpen(false)`

Is seems so simple, but it took me ages to figure out how to get this to work properly -- I tried several different approaches. 😡